javascripts/discourse/components/discourse-category-banners.js

```diff
@@ -78,21 +73,28 @@ export default class DiscourseCategoryBanners extends Component {
   }
 
   #checkTargetCategory(categories) {
-    const currentCategoryName = this.category.name.toLowerCase();
-    const parentCategoryName = this.category.parentCategory
+    const currentCategoryName = this.category?.name.toLowerCase();
+    const parentCategoryName = this.category?.parentCategory
       ? this.category.parentCategory.name.toLowerCase()
       : null;
 
     return (
       Object.keys(categories).length === 0 ||
       categories[currentCategoryName] === "all" ||
       categories[currentCategoryName] === "no_sub" ||
-      (this.category.parentCategory &&
+      (this.category?.parentCategory &&
         (categories[parentCategoryName] === "all" ||
           categories[parentCategoryName] === "only_sub"))
     );
   }
 
+  @action
+  teardownComponent() {
+    document.body.classList.remove("category-header");
+    this.category = null;
+    this.categoryBannerPresence.isPresent = false;
```

Comment:
Nitpick, but with services, it might be better to have methods in the service that set a value to the property, and then in components we can call: `this.categoryBannerPresence.setTo(true)` (or false). It might be a slightly cleaner way to separate concerns.